### PR TITLE
Add `webhooks_notifications` on task levels in `databricks_job` resource

### LINKED ISSUE #2951: [ISSUE] webhook_notifications are not supported at task level
Hey there,

webhook_notifications block is not allowed at task level when using terraform. Though this feature is already supported via databricks UI and the api(https://docs.databricks.com/api/workspace/jobs/create).

It would be good to have webhook notifications on individual tasks of a job.

### PR DESCRIPTION
**Pull request by alexott**
## Changes

Also, replaced email notifications with corresponding structs from Go SDK to make it matching to the API specification, and updated docs correspondingly

This fixes #2951

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [ ] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [ ] using Go SDK

### CODE REVIEW

File: jobs/resource_job.go [hunk 1]
```diff
@@ -118,16 +118,6 @@ type RunJobTask struct {
 	JobParameters map[string]string `json:"job_parameters,omitempty"`
 }
 
-// EmailNotifications contains the information for email notifications after job or task run start or completion
```
Comment (mgyucht) [inline jobs/resource_job.go:121]: Woohoo! Thanks

File: jobs/resource_job.go [hunk 2]
```diff
@@ -208,7 +198,8 @@ type JobTaskSettings struct {
 	// ConditionTask is in private preview
 	ConditionTask *jobs.ConditionTask `json:"condition_task,omitempty" tf:"group:task_type"`
 
-	EmailNotifications     *EmailNotifications            `json:"email_notifications,omitempty" tf:"suppress_diff"`
+	EmailNotifications     *jobs.TaskEmailNotifications   `json:"email_notifications,omitempty" tf:"suppress_diff"`
+	WebhookNotifications   *WebhookNotifications          `json:"webhook_notifications,omitempty" tf:"suppress_diff"`
```
Comment (mgyucht) [inline jobs/resource_job.go:202]: Can we not use the SDK for Webhook Notifications? If you need the Sort() method, I was thinking something like
```go
type WebhookNotifications struct {
  *jobs.WebhookNotifications
}

func (w *WebhookNotifications) Sort() {
}
```
could do the trick

**Review by mgyucht — APPROVED** [review]
LGTM, one small question

File: jobs/resource_job.go (hunk 2 above)
Comment (alexott) [reply jobs/resource_job.go:202]: I'm thinking if we ever need sorting - I think that we need to ask Jobs team to check if they are sorting the submitted values?

Comment (alexott) [reply jobs/resource_job.go]: @mgyucht we still need to sort, but I reimplemented it a bit differently - please look to updates

**Review by mgyucht — APPROVED** [review]
That's perfect, thanks!

**Pull request merged**